### PR TITLE
updated l2vpn l2ckt changes

### PR DESCRIPTION
**Pull request by nchiru**
added xslt in L2CircuitConnectionInfo.yml
added enum for l2ckt connection-status
updated rule properties with supported platforms
removed the playbook

### CODE REVIEW

File: juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule [hunk 1]
```diff
@@ -75,22 +97,94 @@
             rule-properties {
                 version 1;
                 contributor juniper;
-                supported-healthbot-version 4.0.0;
+                supported-healthbot-version 4.3.0;
                 supported-devices {
+                    sensors isis-oc;
```
Comment (gvarao) [inline juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule]: Change the sensor name to actual "l2ckt"

File: juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule [hunk 2]
```diff
@@ -75,22 +97,94 @@
...
                     juniper {
-                         operating-system junos {
+                        operating-system junos {
+                            sensors isis-oc-junos;
                             products MX {
-                                platforms All {
-                                    releases 15.1R1 {
+                                sensors isis-oc-junos;
```
Comment (gvarao) [inline juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule]: Change the sensor name to actual "l2ckt"

File: juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule [hunk 3]
```diff
@@ -75,22 +97,94 @@
...
+                                platforms MX304 {
+                                    releases 23.2 {
+                                        sensors isis-oc-junos;
```
Comment (gvarao) [inline juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule]: Change the sensor name to actual "l2ckt"

File: juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule [hunk 4]
```diff
@@ -75,22 +97,94 @@
...
+                                        release-support min-supported-release;
+                                    }
+                                    releases 22.2 {
+                                        sensors isis-oc;
```
Comment (gvarao) [inline juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule]: Change the sensor name to actual "l2ckt"

File: juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule [hunk 5]
```diff
@@ -75,22 +97,94 @@
...
+                                        release-support min-supported-release;
+                                    }
+                                }
+                                platforms MX204 {
+                                    releases 23.2 {
+                                        sensors isis-oc-junos;
```
Comment (gvarao) [inline juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule]: Change the sensor name to actual "l2ckt"

File: juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule [hunk 6]
```diff
@@ -75,22 +97,94 @@
...
+                                        release-support min-supported-release;
+                                    }
+                                    releases 22.2 {
+                                        sensors isis-oc;
```
Comment (gvarao) [inline juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule]: Change the sensor name to actual "l2ckt"

File: juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule [hunk 7]
```diff
@@ -75,22 +97,94 @@
...
                                 }
                             }
-                         }
-                         operating-system junosEvolved {
+                        }
+                        operating-system junosEvolved {
                             products ACX {
-                                platforms All {
-                                    releases 22.1R1 {
+                                sensors isis-oc;
```
Comment (gvarao) [inline juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule]: Change the sensor name to actual "l2ckt"

File: juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule [hunk 8]
```diff
@@ -75,22 +97,94 @@
...
+                                platforms ACX7024 {
+                                    releases 23.2R1 {
+                                        sensors isis-oc-junos;
```
Comment (gvarao) [inline juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule]: Change the sensor name to actual "l2ckt"

File: juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule [hunk 9]
```diff
@@ -75,22 +97,94 @@
...
+                                        release-support min-supported-release;
+                                    }
+                                    releases 22.3R1 {
+                                        sensors isis-oc;
```
Comment (gvarao) [inline juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule]: Change the sensor name to actual "l2ckt"

File: juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule [hunk 10]
```diff
@@ -75,22 +97,94 @@
...
+                                        release-support min-supported-release;
+                                    }
+                                }
+                            }
+                            products PTX {
+                                sensors isis-oc-junos;
```
Comment (gvarao) [inline juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule]: Change the sensor name to actual "l2ckt"

File: juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule [hunk 11]
```diff
@@ -75,22 +97,94 @@
...
+                                        release-support min-supported-release;
+                                    }
+                                    releases 22.3R1 {
+                                        sensors isis-oc;
+                                        release-support min-supported-release;
+                                    }
+                                }
+                                platforms ACX7509 {
+                                    releases 23.2R1 {
+                                        sensors isis-oc-junos;
```
Comment (gvarao) [inline juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule]: Change the sensor name to actual "l2ckt"

File: juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule [hunk 12]
```diff
@@ -75,22 +97,94 @@
...
+                                        release-support min-supported-release;
+                                    }
+                                    releases 22.3R1 {
+                                        sensors isis-oc;
+                                        release-support min-supported-release;
+                                    }
+                                }
+                                platforms ACX7024X {
+                                    releases 23.2R1 {
+                                        sensors isis-oc-junos;
```
Comment (gvarao) [inline juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule]: Change the sensor name to actual "l2ckt"

File: juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule [hunk 13]
```diff
@@ -75,22 +97,94 @@
...
+                                        release-support min-supported-release;
+                                    }
+                                }
+                                platforms ACX7348 {
+                                    releases 23.2R1 {
+                                        sensors isis-oc-junos;
+                                        release-support min-supported-release;
+                                    }
+                                    releases 22.3R1 {
+                                        sensors isis-oc;
```
Comment (gvarao) [inline juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule]: Change the sensor name to actual "l2ckt"

File: juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule [hunk 14]
```diff
@@ -75,22 +97,94 @@
...
+                                        release-support min-supported-release;
+                                    }
+                                }
+                                platforms ACX7100 {
+                                    releases 23.2R1 {
+                                        sensors isis-oc-junos;
+                                        release-support min-supported-release;
+                                    }
+                                    releases 22.3R1 {
+                                        sensors isis-oc;
```
Comment (gvarao) [inline juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule]: Change the sensor name to actual "l2ckt"

File: juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule [hunk 15]
```diff
@@ -75,22 +97,94 @@
...
+                                        release-support min-supported-release;
+                                    }
+                                }
+                                platforms ACX7100 {
+                                    releases 23.2R1 {
+                                        sensors isis-oc-junos;
```
Comment (gvarao) [inline juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule]: Change the sensor name to actual "l2ckt"

**Review by gvarao — CHANGES_REQUESTED** [review]

File: juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule (hunk 1 above)
Comment (nchiru) [reply juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule]: updated with l2ckt

File: juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule (hunk 2 above)
Comment (nchiru) [reply juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule]: updated with l2ckt

File: juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule (hunk 3 above)
Comment (nchiru) [reply juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule]: updated with l2ckt

File: juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule (hunk 4 above)
Comment (nchiru) [reply juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule]: updated with l2ckt

File: juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule (hunk 5 above)
Comment (nchiru) [reply juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule]: updated with l2ckt

File: juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule (hunk 11 above)
Comment (nchiru) [reply juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule]: updated with l2ckt

File: juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule (hunk 12 above)
Comment (nchiru) [reply juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule]: updated with l2ckt

File: juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule (hunk 13 above)
Comment (nchiru) [reply juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule]: updated with l2ckt

File: juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule (hunk 15 above)
Comment (nchiru) [reply juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule]: updated with l2ckt

File: juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule (hunk 14 above)
Comment (nchiru) [reply juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule]: updated with l2ckt

File: juniper_official/vpn/l2vpn/L2CircuitConnectionInfo.yml [hunk 16]
```diff
@@ -0,0 +1,36 @@
...
+        connection-status: connection-status
+
+xslt: "<xsl:stylesheet version=\"1.0\"
+ xmlns:xsl=\"http://www.w3.org/1999/XSL/Transform\"
+ xmlns:ns=\"some:ns\">
+ <xsl:output omit-xml-declaration=\"yes\" indent=\"yes\"/>
+ <xsl:strip-space elements=\"*\"/>
+ <ns:WhiteList>
+      <name>connection-id</name>
+      <name>connection-status</name>
```
Comment (gaddevijay) [inline juniper_official/vpn/l2vpn/L2CircuitConnectionInfo.yml:26]: neighbor-address is not required?

File: juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule [hunk 17]
```diff
@@ -75,22 +97,94 @@
...
                             products MX {
-                                platforms All {
-                                    releases 15.1R1 {
+                                sensors l2ckt;
+                                platforms MX304 {
+                                    releases 23.2 {
+                                        sensors l2ckt;
+                                        release-support min-supported-release;
+                                    }
+                                    releases 22.2 {
```
Comment (gaddevijay) [inline juniper_official/vpn/l2vpn/check-l2circuit-pw-state.rule:109]: Why are 2 release required for one sensor?

**Review by gaddevijay — CHANGES_REQUESTED** [review]

File: juniper_official/vpn/l2vpn/L2CircuitConnectionInfo.yml (hunk 16 above)
Comment (gaddevijay) [reply juniper_official/vpn/l2vpn/L2CircuitConnectionInfo.yml:26]: u removed connection-id also? Did you test with these changes?

**Comment by sharanyabhat** [comment]
Jenkins please retry a build

**Review by gaddevijay — APPROVED** [review]

**Review by gvarao — APPROVED** [review]

**Pull request merged**